### PR TITLE
Enable stream import for COCO data format

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary

- ticket no. 114761
- Implement `COCOPageMapper`
- Enable the COCO extractor to generate `DatasetItem` using `COCOPageMapper` in the stream manner

### How to test
I parametrized the existing tests for this change.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [x] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [x] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: tests/unit/data_formats/coco/test_roboflow.py [hunk 1]
```diff
@@ -2,12 +2,15 @@
 #
 # SPDX-License-Identifier: MIT
 
+from typing import Any, Dict, Optional, Type
```
Comment (jihyeonyi) [inline tests/unit/data_formats/coco/test_roboflow.py]: Type is imported but not used. Doesn't the commit-hook apply to the test code?

Comment (vinnamkim) [reply tests/unit/data_formats/coco/test_roboflow.py]: The "tests" directory is excluded during the `ruff` check.

Comment (vinnamkim) [reply tests/unit/data_formats/coco/test_roboflow.py]: Anyway I'll revert it with fixing the errors on Windows

**Review by jihyeonyi — APPROVED** [review]

File: src/datumaro/plugins/data_formats/coco/page_mapper.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+@dataclass
+class BraceStatus:
+    flush: bool = False
+    level: int = 0
+    get_started: bool = False
+
+
+@dataclass
+class CurlyStatus:
```
Comment (wonjuleee) [inline src/datumaro/plugins/data_formats/coco/page_mapper.py]: Could you figure out what `BraceStatus` and `CurlyStatus` mean?

File: src/datumaro/plugins/data_formats/coco/page_mapper.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    flush: bool = False
+    level: int = 0
+    buffer: bytes = b""
+    start: int = -1
+    end: int = -1
+
+    def reset(self):
+        self.flush = 0
```
Comment (wonjuleee) [inline src/datumaro/plugins/data_formats/coco/page_mapper.py]: Minor comment: `self.flush = False` for `reset`

File: src/datumaro/plugins/data_formats/coco/page_mapper.py (hunk 2 above)
Comment (vinnamkim) [reply src/datumaro/plugins/data_formats/coco/page_mapper.py]: https://github.com/openvinotoolkit/datumaro/pull/1091/commits/ab88cf9b2462dbd8976b25bb810931cab273ba46 renames `BraceStatus -> BracketStatus` and `CurlyStatus -> BraceStatus`. It also adds docstrings to them.

**Review by wonjuleee — APPROVED** [review]
Looks great!

File: src/datumaro/plugins/data_formats/coco/page_mapper.py (hunk 3 above)
Comment (vinnamkim) [reply src/datumaro/plugins/data_formats/coco/page_mapper.py]: Fixed in 3a8af6dd3196c2d8fc30c4b9029778de1070bfaa

**Pull request merged**